actors/builtin/market/market_actor.go

```diff
@@ -156,22 +157,30 @@ func (a Actor) PublishStorageDeals(rt Runtime, params *PublishStorageDealsParams
 		proposals := AsDealProposalArray(adt.AsStore(rt), st.Proposals)
 		dbp := AsSetMultimap(adt.AsStore(rt), st.DealIDsByParty)
 		// All storage proposals will be added in an atomic transaction; this operation will be unrolled if any of them fails.
+
+		_, worker := builtin.RequestMinerControlAddrs(rt, params.Deals[0].Proposal.Provider)
```

Comment:
Check that deals is non-empty before here (your call whether to succeed vacuously or abort).

Reply:
Eh, I guess. Since we're using panics for everything we would just catch this, but yeah. cleaner to explicitly check.

Reply:
Actor code itself should *never* panic.